deploy/examples/cr-istio.yaml

```diff
@@ -209,3 +209,99 @@ spec:
   resources:
     requests:
       storage: 1Gi
+
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: vault
+  namespace: vault
+
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vault-secrets
+  namespace: vault
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    verbs:
+      - "*"
+
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vault-secrets
+  namespace: vault
+roleRef:
+  kind: Role
+  name: vault-secrets
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: vault
+
+---
+# This binding allows the deployed Vault instance to authenticate clients
+# through Kubernetes ServiceAccounts (if configured so).
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: vault-auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+  - kind: ServiceAccount
+    name: vault
+    namespace: vault
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  namespace: app
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: app
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: app
+      annotations:
+        vault.security.banzaicloud.io/vault-addr: "https://vault.vault:8200"
+        vault.security.banzaicloud.io/vault-tls-secret: vault-tls
+    spec:
+      # initContainers only works if Vault is having PERMISSIVE authentication policy in Istio
+      initContainers:
+        - name: init-ubuntu
+          image: ubuntu
+          command:
+            [
+              "sh",
+              "-c",
+              "echo $AWS_SECRET_ACCESS_KEY && echo initContainers ready",
+            ]
+          env:
+            - name: AWS_SECRET_ACCESS_KEY
+              value: vault:secret/data/accounts/aws#${.AWS_SECRET_ACCESS_KEY} # Go templates are also supported with ${} delimiters
+      containers:
+        - name: app
+          image: alpine
+          command:
+            [
+              "sh",
+              "-c",
+              "echo $AWS_SECRET_ACCESS_KEY && echo going to sleep... && sleep 10000",
+            ]
+          env:
+            - name: AWS_SECRET_ACCESS_KEY
+              value: vault:secret/data/accounts/aws#AWS_SECRET_ACCESS_KEY
```

Comment:
You could delete this as well, there are similar deployment examples in /test/deploy, also applying this whole file at once would not make sense as we need the Vault instance to be running first, then start the webhook and only after create deployments with such annotations.